jahsieAult/homework2Answers.md

```diff
@@ -0,0 +1,308 @@
+### Homework #2
+
+Due Monday, 1/19
+
+Each problem below lists an estimate of its relative difficulty and how much time it requires as a percentage of the whole set.  The total time estimate for this homework set is around 10-15 hours.
+
+Please also read the [learning objectives](objectives.md) for this week.
+
+---
+
+**1)** _(Easy, 10% of total time)_
+
+Revisit your solution to homework #1, problem 5, which expresses an improper fraction as a proper one, and turn your solution into a function.
+
+**a)** Write a function `fractionString(n,d)` which takes 2 parameters (n,d) and returns a string.  For example, 
+`fractionString(7,4)` should return "1 3/4", and `fractionString(3,3)` should (for now) return "1 0/3".  As before, assume that _n_ and _d_ are both positive integers.
+
+```
+function fractionString(n,d) {
+var integer = (Math.floor(n/d)).toString(),
+    numerator = (n % d).toString(),
+    denominator = d.toString(),
+    finalValue = integer + " " + numerator + "/" + denominator;
+	return finalValue;
+}
+
+```
+
+Note that _returning_ a string is not the same as _printing_ a string.  You may use `console.log()` for debugging, but your function should have the correct string as its return value.
+You can also print your return value with an expression like `console.log(fractionString(7,4))`.
+
+**b)** Write a second version of your function which improves the output in two special cases:
+
+- A output string like "0 1/2" should be simplified to just "1/2";
+- A output string like "1 0/3" should be simplified to just "1".
+
+```
+function fractionString(n,d) {
+	var integer = (Math.floor(n/d)).toString(),
+    numerator = (n % d).toString(),
+    denominator = d.toString();
+    if (integer == 0) {
+    	finalValue = numerator + "/" + denominator;
+	}
+	else if (numerator == 0) {
+		finalValue = integer;
+	}
+	else {
+		finalValue = integer + " " + numerator + "/" + denominator;
+	}
+	return finalValue;
+}
```

Comment:
1. Yep! You could simplify a bit by removing the `toString()`s, since autoconversion will turn the numbers into strings when you start adding in the space and the /, but that's all I'd have for that.